### PR TITLE
ci: merge publish workflow with build one

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/squidfunk/mkdocs-material/pull/4994

Merge publish workflow with build one so we can build python package and docker image on pull request to make sure there is no regression.

Also use [`docker/metadata-action`](https://github.com/docker/metadata-action) to generate tags and labels and don't set latest tag on prerelease.

In a follow-up I will work on making the Docker image multi-platform without build time penalty.

### CODE REVIEW

**Comment by squidfunk** [comment]
Thank you! I need a little time to review this PR as we currently need to handle the situation with GitHub Sponsors.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -61,3 +66,89 @@ jobs:
...
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Check Docker image
+        working-directory: /tmp
+        run: |
+          docker run --rm -i -v ${PWD}:/docs ${{ github.event.repository.full_name }}:${{ steps.meta.outputs.version }} new .
+          docker run --rm -i -v ${PWD}:/docs ${{ github.event.repository.full_name }}:${{ steps.meta.outputs.version }} build
+
+      - name: Publish Docker image
+        uses: docker/build-push-action@v3
```
Comment (squidfunk) [inline .github/workflows/build.yml:149]: Can we upgrade this to v4? That was my initial question in #4994 😅 Otherwise dependabot will complain.

Comment (crazy-max) [reply .github/workflows/build.yml:149]: Oh I forgot about this, let me make the change.

Comment (crazy-max) [reply .github/workflows/build.yml]: Done, tested here: https://github.com/crazy-max/mkdocs-material/actions/runs/4274609950/jobs/7441347841

Comment (squidfunk) [reply .github/workflows/build.yml]: Perfect! I will merge this now and then we can test-drive it pushing out 9.0.15. Really looking forward to arm64 support!

**Review by squidfunk — APPROVED** [review]
LGTM!

File: .github/workflows/build.yml (hunk 1 above)
Comment (crazy-max) [reply .github/workflows/build.yml]: Looks good: https://github.com/squidfunk/mkdocs-material/actions/runs/4274651507/jobs/7441421218#step:9:218

**Comment by squidfunk** [comment]
All good:
https://github.com/squidfunk/mkdocs-material/actions/runs/4274651507

Now the only thing that is needed would be to build images for other architectures on a release event. I guess we could make the architectures to build for configurable via environment variable or something similar. Building for `arm64` so support M1 and M2 Macs should be sufficient, I don't think we need `arm/v7`, or do we?

**Comment by crazy-max** [comment]
> I don't think we need `arm/v7`, or do we?

I don't know many people developing on this platform but doesn't hurt to have it too.

> Prior attempt:
> 
>     *

It needs more changes in the Dockerfile than adding extra platforms. I will carry that. Keep you posted.

**Comment by squidfunk** [comment]
Aight, thanks for your work and time invested!

**Pull request merged**